### PR TITLE
Update ReadSourceData to extract RR fields --> Insert into eICR

### PR DESCRIPTION
**Pull request by georgehager**
### tl:dr
Extracts required fields from an RR and inserts them into the eICR

### Suggested Changes

- Add a `rr_to_ecr` function to process the rr and ecr. Extract these fields per @BradySkylight's guidance:
  - "templateId",
  - "id",
  - "code",
  - "title",
  - "effectiveTime",
  - "confidentialityCode",
  - "entry"  with type="DRIV" and "organizer" child with classCode="CLUSTER" and moodCode="EVN"
- Insert these fields into a new <section> at the root level of the ecr
- Make sure the ecr has a `xmlns:xsi` tag so rr fields can be properly parsed later on, and finally:
- In the main function, we still send one `ecr` object in messages to the pipeline; however, we perform the `rr_to_ecr` extract/insert on the `rr` and `ecr` retreived from blob storage before sending the `ecr` object forward

### Testing

Automated Testing
- Modified test to have properly mocked xml for rr and ecr
- Added a test to confirm required fields are extracted from the RR then inserted into the eICR

Manual Testing
- Deployed branch to `test` env in Azure
- Uploaded rr and ecr xml files to blob storage
- Confirmed ReadSourceData function app success

### Notes

- Eventually we want to pull this extract/insert functionality out to a new building block. For now, we made the decision as a first iteration to put the code directly into the ReadSourceData function.
- Failure in the pipeline on the upload to fhir step is not related to rr, it's a known problem

### CODE REVIEW

**Review by DanPaseltiner — APPROVED** [review]
LGTM! Thanks for putting this together @georgehager. I'd like us to add a doc string to the `rr_to_ecr()` function, but don't feel it needs to be blocking.

File: serverless-functions/ReadSourceData/__init__.py [hunk 1]
```diff
@@ -168,3 +168,71 @@ def get_reportability_response(
         reportability_response = ""
 
     return reportability_response
+
+
+# extract rr fields and insert them into the ecr
+def rr_to_ecr(rr, ecr):
```
[the remark this answers is not in the document]
Comment (georgehager) [reply serverless-functions/ReadSourceData/__init__.py:174]: ✅

**Review by BradySkylight — CHANGES_REQUESTED** [review]
We need to add one more field that is currently missing, from the RR into the eICR, the Status.  I missed this when we were doing some digging on how to find/pull the fields from the eICR that were required.

**Comment by georgehager** [comment]
> We need to add one more field that is currently missing, from the RR into the eICR, the Status. I missed this when we were doing some digging on how to find/pull the fields from the eICR that were required.

Thanks for adding this @BradySkylight! Looks like you fixed the failing test issue - I can deploy and test in Azure if we want one more check before moving forward?

**Review by BradySkylight — APPROVED** [review]
Added some new code to handle the 'status' from RR.... as long as George and others are ok with this addition - I approve of all other sections.

**Comment by BradySkylight** [comment]
> > We need to add one more field that is currently missing, from the RR into the eICR, the Status. I missed this when we were doing some digging on how to find/pull the fields from the eICR that were required.
> 
> Thanks for adding this @BradySkylight! Looks like you fixed the failing test issue - I can deploy and test in Azure if we want one more check before moving forward?

That would be nice if you could do that!

**Comment by georgehager** [comment]
Final test in Azure succeeded so I'll go ahead and merge this now @DanPaseltiner @BradySkylight
